templates/web/pages/product_edit/product_edit_form_display.tt.js

```diff
@@ -40,61 +40,44 @@ function show_warning(should_show, nutrient_id, warning_message){
 		\$('#nutriment_'+nutrient_id).css("background-color", "rgb(255 237 235)");
 		\$('#nutriment_question_mark_'+nutrient_id).css("display", "inline-table");
 		\$('#nutriment_sugars_warning_'+nutrient_id).text(warning_message);
-	}else {
-		\$('#nutriment_'+nutrient_id).css("background-color", "white");
-		\$('#nutriment_question_mark_'+nutrient_id).css("display", "none");
+	} else {
+		// clear the warning only if the warning message we don't show is the same as the existing warning
+		// so that we don't remove a warning on sugars > 100g if we change carbohydrates
+		if (warning_message == \$('#nutriment_sugars_warning_'+nutrient_id).text()) {
+			\$('#nutriment_'+nutrient_id).css("background-color", "white");
+			\$('#nutriment_question_mark_'+nutrient_id).css("display", "none");
+		}
 	}
 }
 
-var sugars_value;
-var carbohydrates_value;
-var saturated_fats_value;
-var fat_value;
-
 var required_nutrients_id = ['energy-kj', 'energy-kcal', 'fat', 'saturated-fat', 'sugars', 'carbohydrates', 'fiber', 'proteins', 'salt', 'sodium', 'alcohol'];
 
 required_nutrients_id.forEach(nutrient_id => {
 	\$('#nutriment_' + nutrient_id).on('input', function() {
-		var nutrient_value = \$(this).val();
-		var is_above_or_below_100 = isNaN(nutrient_value) || nutrient_value < 0 || nutrient_value > 100;
-		show_warning(is_above_or_below_100, nutrient_id, "Please enter a value between 0 and 100");
 
-		var crutial_nutrients = ['fat', 'saturated-fat', 'sugars', 'carbohydrates'];
+		// check the changed nutrient value
+		var nutrient_value = \$(this).val().replace(',','.');
+		var is_above_or_below_100 = (isNaN(nutrient_value) && nutrient_value != '-') || nutrient_value < 0 || nutrient_value > 100;
+		// if the nutrition facts are indicated per serving, the value can be above 100
+		if ((nutrient_value > 100) && (\$('#nutrition_data_per_serving').is(':checked'))) {
```

Comment:
can't we check that it's below serving size ?

Reply:
It's tricky, because serving size is parsed on the server (it can be specified with different units etc.), but the user can change it in the form while editing nutrients.